clang/lib/AST/ExprConstant.cpp

```diff
@@ -8480,6 +8480,54 @@ class LValueExprEvaluator
 };
 } // end anonymous namespace
 
+/// Get an lvalue to a field of a lambda's closure type.
+static bool GetLambdaCaptureAsLValue(EvalInfo &Info, const Expr *E,
+                                     LValue &Result, const CXXMethodDecl *MD,
+                                     const FieldDecl *FD,
+                                     bool LValueToRValueConversion) {
+  // Static lambda function call operators can't have captures. We already
+  // diagnosed this, so bail out here.
+  if (MD->isStatic()) {
+    assert(Info.CurrentCall->This == nullptr &&
+           "This should not be set for a static call operator");
+    return false;
+  }
+
+  // Start with 'Result' referring to the complete closure object...
+  if (MD->isExplicitObjectMemberFunction()) {
+    // Self may be passed by reference or by value.
+    auto *Self = MD->getParamDecl(0);
+    if (Self->getType()->isReferenceType()) {
+      APValue *RefValue = Info.getParamSlot(Info.CurrentCall->Arguments, Self);
+      Result.setFrom(Info.Ctx, *RefValue);
+    } else {
+      auto *VD = Info.CurrentCall->Arguments.getOrigParam(Self);
+      auto *Frame =
+          Info.getCallFrameAndDepth(Info.CurrentCall->Arguments.CallIndex)
+              .first;
+      auto Version = Info.CurrentCall->Arguments.Version;
+      Result.set({VD, Frame->Index, Version});
```

Comment:
Not entirely sure this is the intended way to do this candidly, but I couldn’t figure out how else I should obtain an lvalue to the parameter here. This is based on `LValueExprEvaluator::VisitVarDecl`, where this entire function was also originally located. Is there perhaps some other helper that I missed?

Reply:
I don't know of a helper to do this (I couldn't spot one when I went looking), so it's good that you factored this out.

Reply:
I could make this a separate helper function, but it’s only used here from what I can tell (the code in `LValueExprEvaluator::VisitVarDecl` where I apparently got this from is quite a bit more complicated and does more iirc), so I personally don’t think it’s really necessary.

Reply:
Sorry, I meant that the current factoring is good (we're able to call this from the two places it matters).

Reply:
Ah, I see now